src/events/etcd.py

```diff
@@ -195,3 +220,60 @@ def update_admin_password(self, admin_secret_id: str) -> None:
                         logger.error(e)
         except (ModelError, SecretNotFoundError) as e:
             logger.error(e)
+
+    def update_private_key(self, private_key_id: str, tls_type: TLSType) -> None:
+        """Update the private key in etcd."""
+        logger.debug("Updating TLS private key.")
+
+        tls_relation = (
+            self.charm.state.peer_tls_relation
+            if tls_type == TLSType.PEER
+            else self.charm.state.client_tls_relation
+        )
+
+        tls_requirer = (
+            self.charm.tls_events.peer_certificate
+            if tls_type == TLSType.PEER
+            else self.charm.tls_events.client_certificate
+        )
+        try:
+            secret_content = get_secret_from_id(self.charm.model, private_key_id).get(
+                "private-key"
+            )
+        except (ModelError, SecretNotFoundError) as e:
+            logger.error(e)
+            return
+
+        if secret_content is None:
```

Comment:
Under which conditions would `secret_content is None` be true, given we are catching a `SecretNotFoundError` previously?